components/script/dom/htmlimageelement.rs

```diff
@@ -1159,6 +1172,40 @@ impl HTMLImageElement {
         }
     }
 
+    // Step 2 for <https://html.spec.whatwg.org/multipage/#dom-img-decode>
+    fn react_to_decode_image_sync_steps(&self) {
+        let document = document_from_node(self);
+        // Step 2.1 of <https://html.spec.whatwg.org/multipage/#dom-img-decode>
+        if !document.is_fully_active() ||
+            matches!(self.current_request.borrow().state, State::Broken)
+        {
+            self.reject_image_decode_promises();
```

Comment:
here, as per the comment below, let's only reject the promise associated with the task, and remove the resolve logic below. Instead, if the promise is not rejected, we should store it on `pending_`, and resolve/reject it later as the request state changes. 

I'm going to run the wpt tests before this change, and after, just to see if this makes a difference.

Reply:
I removed the resolve logic but it broke lot of tests.

Reply:
Ok, I will just run the tests once to see, but probably we should then keep it. I guess its a kind of race condition in the spec, between the microtask and the parallel steps.